src/dbnode/network/server/tchannelthrift/cluster/service.go

```diff
@@ -406,3 +397,13 @@ func (s *service) Truncate(tctx thrift.Context, req *rpc.TruncateRequest) (*rpc.
 	res.NumSeries = truncated
 	return res, nil
 }
+
+func toRPCError(err error) *rpc.Error {
+	if client.IsResourceExhaustedError(err) {
+		return tterrors.NewResourceExhaustedError(err)
+	}
+	if client.IsBadRequestError(err) {
+		return tterrors.NewBadRequestError(err)
+	}
+	return tterrors.NewInternalError(err)
+}
```

Comment:
Can we put this in the `convert` package?

In fact this method already exists, we would just need to add it:
https://github.com/m3db/m3/blob/master/src/dbnode/network/server/tchannelthrift/convert/convert.go#L187:L197

Reply:
https://github.com/m3db/m3/pull/3005/commits/fba214e74c2b4ebc76db8bd0ae2ce8e46d0c9abc